source/IrcDotNet/IrcUtilities.cs

```diff
@@ -1,12 +1,15 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Text.RegularExpressions;
 using IrcDotNet.Properties;
 
 namespace IrcDotNet
 {
     // Utilities for IRC.
     internal static class IrcUtilities
     {
+        private static readonly Regex ValidTagKeyRegex = new Regex(@"^([\w.-]+/)?[\w-]+$");
```

Comment:
You need to allow an optional leading plus to allow client prefixes.

Reply:
Thanks for reviewing @kijewski